lib/utils.js

```diff
@@ -494,19 +494,23 @@ class Utils {
         return __awaiter(this, void 0, void 0, function* () {
             const outputDir = Utils.getJobOutputDirectoryPath();
             let markdownContent = '';
+            const sectionContents = {};
+            // Read all sections.
             for (const sectionName of Utils.JOB_SUMMARY_MARKDOWN_SECTIONS_NAMES) {
                 const fullPath = path.join(outputDir, sectionName, 'markdown.md');
-                // Check file exists
-                if (!(0, fs_1.existsSync)(fullPath)) {
-                    continue;
+                if ((0, fs_1.existsSync)(fullPath)) {
+                    sectionContents[sectionName] = yield Utils.readSummarySection(fullPath, sectionName);
                 }
-                markdownContent += yield Utils.readSummarySection(fullPath, sectionName);
             }
-            // No section was added, return empty string
-            if (markdownContent == '') {
-                return '';
+            // If build info was published, remove generic upload section to avoid duplications with generic modules.
+            if (sectionContents[MarkdownSection.BuildInfo] != '') {
+                sectionContents[MarkdownSection.Upload] = '';
```

Comment:
Are we removing the "generic files" section from the Job Summary if build-info was published?
If so, why can't we leave the "generic files" section as is, and just add the published packages (which aren't generic), if build-info was published?